domain.go

```diff
@@ -1832,10 +1832,11 @@ func (d *Domain) ListAllInterfaceAddresses(src uint) ([]DomainInterface, error)
 
 		for k := 0; k < numAddr; k++ {
 			ifaces[i].Addrs[k] = DomainIPAddress{}
-			ifaces[i].Addrs[k].Type = int(addrSlice[k]._type)
-			ifaces[i].Addrs[k].Addr = C.GoString(addrSlice[k].addr)
-			ifaces[i].Addrs[k].Prefix = uint(addrSlice[k].prefix)
-
+			if addSlice[k] != nil {
```

Comment:
This is part of the stack trace I get when not checking if it's `nil`. There's a NPE in domain.go:1835.

```panic: runtime error: invalid memory address or nil pointer dereference
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x585c1d]
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: goroutine 73 [running]:
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/libvirt/libvirt-go.(*Domain).ListAllInterfaceAddresses(0xc4203b6030, 0x1, 0x1, 0xc4203c5db0, 0xc42006ad00, 0x
c4203e2040, 0xc4203b5ed0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/libvirt/libvirt-go/domain.go:1835 +0x35d
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/libvirt.getDomainInterfacesViaQemuAgent(0x7f1954001280, 0x1, 0x0, 0x0, 0xc4203b6098)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt/qemu_agent.go:19 +0xd9
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/libvirt.getDomainInterfaces(0xc4203b6430, 0x44, 0xc4203b6240, 0x1, 0x1, 0x0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt/resource_libvirt_domain.go:971 +0x4b
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/libvirt.waitForNetworkAddresses(0xc4203620a0, 0x1, 0x1, 0x7f1954001280, 0x1, 0x11eede0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt/resource_libvirt_domain.go:909 +0x243
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/libvirt.resourceLibvirtDomainCreate(0xc4202e13b0, 0xaa0b60, 0xc42025f000, 0x0, 0x0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt/resource_libvirt_domain.go:485 +0x37ea
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc42030a420, 0xc4202e2910, 0xc42039c8c0, 0xaa0b60, 0xc42
025f000, 0x1, 0x28, 0xc4202d50b0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:186 +0x48d
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc42030c0e0, 0xc4202e28c0, 0xc4202e2910, 0xc42039c8c0, 0
x7f19801134b0, 0x0, 0x0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:242 +0x9b
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc420329380, 0xc42039c840, 0xc420399080, 0x0, 0x0
)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /home/thomas/go/src/github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:488 +0x57
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: reflect.Value.call(0xc42024e2a0, 0xc42025edf8, 0x13, 0xbd4988, 0x4, 0xc4202bef20, 0x3, 0x3, 0x44d28c, 0xc4202aa738, ...)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /usr/local/go/src/reflect/value.go:434 +0x91f
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: reflect.Value.Call(0xc42024e2a0, 0xc42025edf8, 0x13, 0xc4202aa720, 0x3, 0x3, 0x0, 0x180000, 0x0)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /usr/local/go/src/reflect/value.go:302 +0xa4
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: net/rpc.(*service).call(0xc420325840, 0xc420325800, 0xc420334780, 0xc42033e000, 0xc420329480, 0xaa1ae0, 0xc42039c840, 0x16, 0xaa1b20, 0xc420399080, ...)
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /usr/local/go/src/net/rpc/server.go:387 +0x144
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt: created by net/rpc.(*Server).ServeCodec
2017/04/07 15:27:19 [DEBUG] plugin: terraform-provider-libvirt:         /usr/local/go/src/net/rpc/server.go:481 +0x404
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalWriteState
2017/04/07 15:27:19 [DEBUG] plugin: /home/thomas/go/bin/terraform-provider-libvirt: plugin process exited
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalIf
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalWriteState
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalWriteDiff
2017/04/07 15:27:19 [DEBUG] root: eval: *terraform.EvalApplyPost
2017/04/07 15:27:19 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:
```